miniapps/solvers/plor_solvers.cpp

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2010-2021, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+//
+//                 ------------------------------------------
+//                 Parallel Low-Order Refined Solvers Miniapp
+//                 ------------------------------------------
+//
+// This miniapp illustrates the use of low-order refined preconditioners for
+// finite element problems defined using H1, H(curl), H(div), or L2 finite
+// element spaces. The following problems are solved, depending on the chosen
+// finite element space:
+//
+// H1 and L2: definite Helmholtz problem, u - Delta u = f
+//            (in L2 discretized using the symmetric interior penalty DG method)
+//
+// H(curl):   definite Maxwell problem, u + curl curl u = f
+//
+// H(div):    grad-div problem, u - grad(div u) = f
+//
+// In each case, the high-order finite element problem is preconditioned using a
+// low-order finite element discretization defined on a Gauss-Lobatto refined
+// mesh. The low-order problem is solved using hypre's AMG preconditioners:
+// BoomerAMG is used for H1 and L2 problems, AMS is used for H(curl) and 2D
+// H(div) problems, and ADS is used for 3D H(div) problems.
+//
+// For vector finite element spaces, the special "Integrated" basis type is used
```

Comment:
Forget my other comments, this is probably the place to cite Dohrmann's paper.

Reply:
Good idea. Done.